.changeset/slow-brooms-camp.md

```diff
@@ -0,0 +1,33 @@
+---
+'@graphql-mesh/soap': patch
+'@omnigraph/json-schema': patch
+'@omnigraph/openapi': patch
+'@graphql-mesh/transport-rest': patch
+'@graphql-mesh/transport-soap': patch
+'@graphql-mesh/types': patch
+---
+
+Introduce a standard Upstream Error Format for HTTP-based sources;
+
+So all sources throw an error will have the extensions in the following format;
+
+```json5
+{
+  "extensions": {
+    "request": { // The details of the request made to the upstream service
+      "endpoint": "https://api.example.com",
+      "method": "GET",
+    },
+    "http": { // The details of the HTTP response from the upstream service
```

Comment:
Since the first field is named `request` I would name this `response`. Is there a reason for seperating the body in its own field ?
```suggestion
    "response": { // The details of the HTTP response from the upstream service
```